pyproject.toml

```diff
@@ -24,6 +24,8 @@ dependencies = [
     "astropy",
     "typing-extensions>=4.3.0",
     "numba",
+    "fsspec",
+    "adlfs",
```

Comment:
This should probably only be a `dev` dependency.

Reply:
fsspec should remain a primary dependency since that is what determines all IO,

I've made adlfs and s3fs dev dependencies